pkg/operator/certrotation/signer.go

```diff
@@ -60,12 +67,19 @@ func (c RotatedSigningCASecret) ensureSigningCertKeyPair(ctx context.Context) (*
 	if apierrors.IsNotFound(err) {
 		// create an empty one
 		signingCertKeyPairSecret = &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: c.Namespace, Name: c.Name}}
+		secretExists = false
 	}
 	signingCertKeyPairSecret.Type = corev1.SecretTypeTLS
 
 	if c.Owner != nil {
 		ensureOwnerReference(&signingCertKeyPairSecret.ObjectMeta, c.Owner)
 	}
+	if NeedsTLSMetadataUpdate(&signingCertKeyPairSecret.ObjectMeta, c.JiraComponent, c.Description) && secretExists {
```

Comment:
Is `NeedsTLSMetdataUpdate` mutating more like `ensureOwnerReference`?  If so, `ensureTLSMetadata`?